header.go

```diff
@@ -1837,6 +2185,37 @@ func (h *ResponseHeader) parse(buf []byte) (int, error) {
 	return m + n, nil
 }
 
+func (h *ResponseHeader) parseTrailer(buf []byte) (int, error) {
+	if buf[0] == '0' {
```

Comment:
Shouldn't this check `if len(buf) > 0`?

Reply:
This was checked by `mustPeekBuffered()` above.
```go
func mustPeekBuffered(r *bufio.Reader) []byte {
	buf, err := r.Peek(r.Buffered())
	if len(buf) == 0 || err != nil {
		panic(fmt.Sprintf("bufio.Reader.Peek() returned unexpected data (%q, %v)", buf, err))
	}
	return buf
}
```